tutorial/write-custom-widget.md

```diff
@@ -107,3 +107,178 @@ Take care not to keep any important state in your renderer - animation tickers
 are well suited to that location but user state would not be. A widget that is
 hidden may have it's renderer destroyed and if it is shown again the new renderer
 must be able to reflect the same widget state.
+
+### Example Widget
+
+The following covers in detail the creation of a simple Widget and it's Renderer.
+
+#### Create a Widget
+
+So the widget requires the`widget.BaseWidget`. For example:
+
+```go
+type MyWidget struct {
+	widget.BaseWidget           // Inherit from BaseWidget
+	minSize           fyne.Size // The minimum size of the widget
+	text              string	// The text to display in the widget
+}
+```
+
+To construct the widget create a New<widgetname> function. The example we are using is called MyWidget so by convention the following constructor function name would be used:
+
+```go
+func NewMyWidget(W, H float32, text string) *MyWidget {
+	w := &MyWidget{ 							 // Initialise the widget storage
+		text:    text,                           // Save the text
+		minSize: fyne.Size{Width: W, Height: H}, // Set the size
+	}
+	w.ExtendBaseWidget(w) // Required to initialise the base widget
+	return w
+}
+```
+
+We need to override the MinSize() function so we can return the minimum size of our widget. 
```

Comment:
This is not really true - the default `MinSize` method will look up the renderer information, which is completed later in the example.